### DISCUSSION TITLE
Keep values loaded from remote at state after populating form with setvalues

### DISCUSSION THREAD

**Discussion by zelucena** [react-hook-form/react-hook-form#1920] [Q&A] [answered]
In my real world application I have several nested fields and conditional rendering, all loaded by a remote call.
I made this sandbox (https://codesandbox.io/s/cool-fire-3jvhy?file=/src/App.js) which shows I initially have a nested property that completely goes missing at some point.
My questions are:
* Do I really need to manually set values after conditional rendering? Why isn't <Controller /> able to manage that?
* How can I prevent my values from disappearing? Should I either use little-state-machine or redux for that purpose?
* Is there any section covering this issue I have missed?

**Comment by zelucena** [comment, marked as answer]
I've realized this might be a duplicate from #1917
I'll try using reset().
I've had some improvement here. There might be some improvement yet to be done
https://codesandbox.io/s/adoring-stallman-mus0s?file=/src/App.js

**Reply by bluebill1049** [reply]
let me know if this resolves your problem.

**Reply by zelucena** [reply]
So far it does. I'm still not sure whether this is the most elegant approach though.

**Reply by bluebill1049** [reply]
what would be the elegant approach?

**Reply by zelucena** [reply]
My only other reference point for React is Formik with controlled fields. As it is, there is no need to manually update values during conditional rendering. I don't quite understand why RHF replaces the state entirely in my first sandbox

**Reply by bluebill1049** [reply]
> I don't quite understand why RHF replaces the state entirely in my first sandbox

I don't quite get your question above.

I quickly did one in case if that's what you mean.
https://codesandbox.io/s/jovial-grass-y6wlm?file=/src/App.js

**Reply by zelucena** [reply]
I just went to my real application and it's all loading properly. I'm still working on masked numbers but I'm aware there's an issue for that. Thanks

**Reply by bluebill1049** [reply]
ah super glad to hear that. hey mask input, you should use `Controller`.

**Reply by zelucena** [reply]
My first impressions of this lib are pretty good. I wish these issues get incorporated to the docs as they seem very recurring.

**Reply by bluebill1049** [reply]
The doc is open-sourced as well, feel free to send small PR to improve on it as well.

**Reply by zelucena** [reply]
Good to know. I might if I'm able to come up with a more useful example
